packages/app-mobile/utils/ShareUtils.ts

```diff
@@ -0,0 +1,35 @@
+import Resource from '@joplin/lib/models/Resource';
+import { ResourceEntity } from '@joplin/lib/services/database/types';
+import shim from '@joplin/lib/shim';
+import { CachesDirectoryPath } from 'react-native-fs';
+
+const DIR_NAME = 'sharedFiles';
+
+/**
+ * Copy a file to be shared to cache, renaming it to its orignal name
+ */
+export async function copyToCache(resource: ResourceEntity): Promise<string> {
+	let filename = resource.filename ?
+		`${resource.filename}.${resource.file_extension}` :
+		resource.title;
+
+	if (!filename) {
+		filename = ['untitled', resource.file_extension].join('.');
+	}
```

Comment:
As we'd like to get rid of the filename field eventually, could the logic be changed so that it uses resource.title first? And actually since this is a cached file, perhaps we can just use a random filename, which in the end would be safer?

Reply:
In practice it's the title that is used but it makes sense to have it as first choice. I'll update.
Agree, it makes sense to use random name, or actually, resource id would be just as good.

Reply:
Turns out there's already a method that does exactly this - `Resource.friendlySafeFilename`